composer/helpers/bin/run

```diff
@@ -5,78 +5,95 @@ declare(strict_types=1);
 
 namespace Dependabot\Composer;
 
-require __DIR__ . '/../vendor/autoload.php';
+require_once __DIR__ . '/../vendor/autoload.php';
 
 // Get details of the process to run from STDIN. It will have a `function`
 // and an `args` method, as passed in by UpdateCheckers::Php
 $request = json_decode(file_get_contents('php://stdin'), true);
 
-// Increase the default memory limit the same way Composer does (but clearer)
-if (function_exists('ini_set')) {
-    $memoryInBytes = function ($value) {
-        $unit = strtolower(substr($value, -1, 1));
-        $value = (int) $value;
-        if ($unit == 'g') {
-            $value *= (1024 * 1024 * 1024);
-        } elseif ($unit == 'm') {
-            $value *= (1024 * 1024);
-        } elseif ($unit == 'k') {
-            $value *= 1024;
-        }
-
-        return $value;
-    };
-
-    $memoryLimit = trim(ini_get('memory_limit'));
-    // Increase memory_limit if it is lower than 1900MB
-    if ($memoryLimit != -1 && $memoryInBytes($memoryLimit) < 1024 * 1024 * 1900) {
-        @ini_set('memory_limit', '1900M');
+$memoryInBytes = static function (string $setting): int {
+    $unit = strtolower(substr($setting, -1, 1));
+    $value = (int) $setting;
+
+    if ($unit === 'g') {
+        return $value * 1024 * 1024 * 1024;
+    }
+
+    if ($unit === 'm') {
+        return $value * 1024 * 1024;
     }
 
-    // Set user defined memory limit
-    if ($memoryLimit = getenv('COMPOSER_MEMORY_LIMIT')) {
-        @ini_set('memory_limit', $memoryLimit);
+    if ($unit === 'k') {
+        return $value * 1024;
     }
-    unset($memoryInBytes, $memoryLimit);
+
+    return $value;
+};
+
+$memoryLimit = trim(ini_get('memory_limit'));
+
+// Increase memory_limit if it is lower than 1900MB
+if ($memoryLimit !== '-1' && $memoryInBytes($memoryLimit) < 1024 * 1024 * 1900) {
+    @ini_set('memory_limit', '1900M');
 }
 
-date_default_timezone_set('Europe/London');
+// Set user defined memory limit
+$memoryLimit = getenv('COMPOSER_MEMORY_LIMIT');
 
-// This storage is freed on error (case of allowed memory exhausted)
-$memory = str_repeat('*', 1024 * 1024);
```

Comment:
Looks like you're removing this, but I think it's important. Previously we were seeing OOM errors fail to get written - allocating some memory here that can be freed up when processing them fixes that.

Reply:
Removed the commit!